Add UI Tests for simple text formatting in Aztec

rachelmcr opened the pull request:
This PR adds a suite of Aztec editor tests to the WordPressUITests target. They test each of the text formatting buttons in the toolbar, plus the `hr` and `more` tags. The tests for lists are currently disabled (pending lists being fully implemented in Aztec) and these tests also exclude the media picker, to be tested separately.

Included with the tests are some new `accessibilityIdentifier` properties, to support running the tests in other languages. (I plan to revisit the header style tests after https://github.com/wordpress-mobile/WordPress-iOS/issues/7005 is closed, to replace the strings used in those tests with accessibility identifiers.)

It would be great to get a review from both the testing side (cc @hoverduck) as well as the iOS development side (cc @catehstn) to make sure everything looks ok. :)

hoverduck commented:
A few notes:

1) I was occasionally seeing a problem with the `typeAndSelectText` function failing to actually highlight the text, which meant it failed to find `Select All`.  I'm not sure if that's a timing issue that needs a little delay after typing and before selecting, or what.

2) Also on that note, I'd recommend making that function generic and moving it to the same place as `clearAndEnterText`, as an extension of XCUIElement.  That way it's not hard-coded to be looking for the richTextField, and could be called on any element.

3) The tests are really slow from logging in and out every time.  I think we should be able to add a `loginIfNeeded()` function that could log us in (and maybe enable Aztec), and then the tearDown function could just discard the draft and exit the editor.  If we don't enable Aztec at login we'd still need to jump over to the App Settings page to verify it's enabled every time, but eventually I assume that'll be the default and could drop that.  Or we could add a check on opening the editor to verify that it sees Aztec, and only navigate out to App Settings if it sees the legacy editor.

Making it robust enough to handle any starting state (logged in/logged out/aztec enabled/aztec disabled) is complicated, but I think it's worthwhile to speed things up.  We could even just make the decision for now to just accept the risk of some unsupported starting state and simplify the implementation to assume the initial login and Aztec enable worked and never double-check it.  It should be a pretty low risk.

4) Part of this is my fault (and this doesn't need to be addressed in this PR), but we should have a consistent naming style for the accessibility IDs.  All of the ones I added previously are in _camelCase_, a few of the ones that were already there are in _AllCapsCase_, and the ones you've added here _are_all_separated_by_underscores_.  I'm not sure if there's a style guideline for iOS on that.

rachelmcr commented:
Thanks for the notes!

> I was occasionally seeing a problem with the typeAndSelectText function failing to actually highlight the text, which meant it failed to find Select All. 

Interesting! I haven't seen that in running the tests locally so I'll see if I can track down the cause and fix that up.

> I'd recommend making that function generic and moving it to the same place as clearAndEnterText, as an extension of XCUIElement. 

Agreed. I've made it generic (and renamed it `enterAndSelectText` to match `clearAndEnterText`) and moved it to be an extension of XCUIElement.

> The tests are really slow from logging in and out every time. I think we should be able to add a loginIfNeeded() function that could log us in (and maybe enable Aztec), and then the tearDown function could just discard the draft and exit the editor.

That definitely makes sense. I was trying to log out so the tests start at more or less the same point each time, but there's no good reason to do the login dance each time. I added a `loginIfNeeded()` function as suggested and removed the logout during teardown.

I left in the Aztec check because once you take out logout/login the Aztec check is pretty quick, and once Aztec is the default we can remove that check.

Let me know if you think that isn't sufficient and I'll make further changes to balance speed and robustness.

> Part of this is my fault (and this doesn't need to be addressed in this PR), but we should have a consistent naming style for the accessibility IDs.

Good point, thanks for mentioning it. I used the Aztec IDs added in https://github.com/wordpress-mobile/WordPress-iOS/pull/6967 to start and then used the same naming style for additional IDs. However, our [style guide](https://github.com/wordpress-mobile/swift-style-guide) points to the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/#general-conventions), which calls for _lowerCamelCase_. I'll fix the new IDs I've added in this PR accordingly.

rachelmcr commented:
After further discussion with @catehstn and @hoverduck we agreed to move these tests to the Aztec repo. Closing this PR — I'll open a new one for Aztec when it's ready.
